### PR TITLE
Structured query with timefilter and index author data request

### PR DESCRIPTION
**Pull request by yang-han**

### CODE REVIEW

**Comment by webroot-kaito** [comment]
As discussed offline, let's do a more gradual rollout to let miners be prepared & use more time to validate our reward mechanism. So I suggest the following -

1. Let first support AND/OR queries. This will be reflected in the structured query API instead of using query string, so that we can build a customized prompt for it.
2. For the default miner, let's just issue an ES search for any structured query (as a new task), and keep the old task the same. And we can gradually ramp up the portion of this task. This way miners running our default code will still have crawlers triggered on the original path to get data, but will also start to be evaluated in a crawler agnostic way.
3. Independent of this change (maybe in a separate PR), let's move validator to use ApiDojo crawler for a more accurate evaluation, and we can provide that crawler to our miner as optional. This will also get us prepared to onboard data from other Subnets.

File: openkaito/evaluation/evaluator.py [hunk 1]
```diff
@@ -219,15 +274,157 @@ def llm_ranking_evaluation(self, query_string, size, docs, retries=3):
...
+Outdated: Time-sensitive information that is no longer current or relevant.
+Off topic: Superficial content lacking depth and comprehensive insights.
+Somewhat Relevant: Offers partial insight but lacks depth and comprehensive coverage.
+Relevant: Comprehensive, insightful content suitable for informed decision-making.""",
+                    },
+                    {
+                        "role": "system",
+                        "content": """You will be given a list of documents with id and you have to rate them based on the relevance to the structued query. 
+The structured query can contain AND/OR operators. 
+AND operator: The document should contain all the keywords in the query.
```
Comment (webroot-kaito) [inline openkaito/evaluation/evaluator.py:325]: Should we make this more semantic? Basically AND means the doc contains meaningful discussions for both A and B, OR means contains meaningful discussions for either A OR B. I think that was reflected in the in-context example, but wondering if we should also focus less on keyword match here.

Comment (yang-han) [reply openkaito/evaluation/evaluator.py]: yes, agree!

**Review by webroot-kaito — APPROVED** [review]
lgtm!

**Pull request merged**